colorizer_data/converter.py

```diff
@@ -0,0 +1,392 @@
+        track_column (str): The name of the column containing track IDs. Defaults to "Track."
+        image_column (str): The name of the column containing filepaths to the segmentation images.
+            Defaults to "File Path." Images will be copied and remapped. If they are 3D, they will
+            be flattened along the Z-axis using a max projection.
+        centroid_x_column (str): The name of the column containing x-coordinates of object
+            centroids, in pixels relative to the frame image, where 0 is the left edge of the
+            image. Defaults to "Centroid X."
+        centroid_y_column (str): The name of the column containing y-coordinates of object
+            centroids, in pixels relative to the frame image, where 0 is the top edge of the image.
+            Defaults to "Centroid X.""
+        outlier_column (str): The name of the column containing outlier flags. 0/False indicates a
+            normal object, while 1/True indicates an outlier. Outliers are excluded from min/max
+            calculation for features. Defaults to "Outlier."
+        backdrop_columns (List[str] | None): A list of column names containing file paths to
+            backdrop images. If set, these images will be copied and included in the dataset as
+            backdrops that can be toggled. Defaults to `None`.
+        backdrop_info (Dict[str, BackdropMetadata] | None): A dictionary mapping column names to
+            `BackdropMetadata` metadata. This includes the backdrop's name, description, and
+            file paths. If the files do not exist in the dataset directory, the files
+            will be copied to it and the paths updated to be relative to the manifest. Defaults to
+            `None`.
+        feature_column_names (List[str] | None): An array of feature column names. If a value is
+            provided, ONLY the provided column names will be parsed as features; otherwise, ALL
+            columns that aren't specified as a backdrop or a data column (e.g. object ID, time,
+            track, etc.) will be treated as a feature. Defaults to `None`.
+        feature_info (Dict[str, FeatureInfo] | None): A dictionary mapping column names to
+            `FeatureInfo` metadata. This includes the feature's type (continuous, discrete, or
+            categorical), units, min/max value overrides, descriptions, and category order for
+            categorical features. If a feature's column name does not exist in the `feature_info`
+            (or the dictionary is `None`), the feature type and metadata will be inferred based
+            on column values. Defaults to `None`.
+        force_frame_generation (bool): If True, frames will be regenerated even if they already
+            exist. If False (default), frames will be regenerated only when changes are detected.
+        use_json (bool): If True, data will be written as JSON files instead of the default Parquet
```

Comment:
Nit: for futureproofing this might be better as a string name or enum type of thing.  output_format = "parquet" or "json".  totally optional... I don't know if or when a third format will show up